src/Core/Services/MetadataProviders/MetadataProviderFactory.cs

```diff
@@ -62,7 +62,9 @@ private void ConfigureMetadataProviders()
 
         public void OnConfigChanged(object? sender, HotReloadEventArgs args)
         {
+            _metadataProviders.Clear();
             ConfigureMetadataProviders();
+            InitializeAsync().GetAwaiter().GetResult();
         }
```

Comment:
Both the changes in this PR seem to be unnecessary with #2428 
After your discussion, which PR did you decide to keep?

Reply:
this one. Aaron will update his PR based on my changes